internal/services/web/function_app.go

```diff
@@ -331,8 +344,8 @@ func getBasicFunctionAppAppSettings(d *pluginsdk.ResourceData, appServiceTier, e
 
 	// On consumption and premium plans include WEBSITE_CONTENT components, unless it's a Linux consumption plan
 	// (see https://github.com/Azure/azure-functions-python-worker/issues/598)
-	if (strings.EqualFold(appServiceTier, "dynamic") || strings.EqualFold(appServiceTier, "elasticpremium") || strings.HasPrefix(strings.ToLower(appServiceTier), "premium")) &&
-		!strings.EqualFold(d.Get("os_type").(string), "linux") {
+	if !(strings.EqualFold(appServiceTier, "dynamic") && strings.EqualFold(d.Get("os_type").(string), "linux")) &&
+		(strings.EqualFold(appServiceTier, "dynamic") || strings.HasPrefix(appServiceTier, "elastic") || strings.HasPrefix(appServiceTier, "Elastic")) {
```

Comment:
Minor but a bit shorter?

```suggestion
		(strings.EqualFold(appServiceTier, "dynamic") || strings.HasPrefix(strings.ToLower(appServiceTier), "elastic")) {
```